### PR TITLE
Add tsc workflow

### LINKED ISSUE #5: Add a type check workflow
Add type check package scripts that run `tsc --pretty --noEmit` on each workspace, and a type check GitHub action workflow.

### PR DESCRIPTION
**Pull request by masonmcelvain**
Closes #5 

pnpm recursive docs: https://pnpm.io/cli/recursive

## QA

Make sure the type check action runs and passes.

Locally, add a type error and make sure the command catches the error.

### CODE REVIEW

**Comment by masonmcelvain** [comment]
QA ✌🏻 Added a type error to `extensions/ui-extension/src/index.tsx` and it was caught

File: .github/actions/pnpm-setup/action.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Setup pnpm
+description: Installs pnpm, and caches the data store
+runs:
+   using: "composite"
+   steps:
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v2.2.4
+        with:
+           version: 8
+           run_install: false
```
Comment (danielcliu-ifixit) [inline .github/actions/pnpm-setup/action.yml:10]: why don't we want to install?

**Review by danielcliu-ifixit — APPROVED** [review]
CR :v:

File: .github/actions/pnpm-setup/action.yml (hunk 1 above)
Comment (masonmcelvain) [reply .github/actions/pnpm-setup/action.yml:10]: I think we need to setup the cache before we run `pnpm install`

The action provider recommends it: https://github.com/pnpm/action-setup#use-cache-to-reduce-installation-time

**Pull request merged**